api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.dao.impl;
+
+import javax.inject.Inject;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.openmrs.Encounter;
+import org.openmrs.api.EncounterService;
+import org.openmrs.module.fhir2.api.dao.FhirEncounterDao;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class FhirEncounterDaoImpl implements FhirEncounterDao {
+	
+	@Inject
+	private EncounterService encounterService;
+	
+	@Override
+	public Encounter getEncounterByUuid(String uuid) {
+		return encounterService.getEncounterByUuid(uuid);
```

Comment:
I tend to be in favour of the `sessionFactory.getCurrentSession().createCriteria(Encounter.class).add(eq("uuid", uuid)).uniqueResult()` pattern.